tests/test_notify_suppliers_with_incomplete_applications.py

```diff
@@ -0,0 +1,218 @@
+#                 'scaling': False, 
+#                 'serviceBenefits': ['user friendly', 'company specific'], 
+#                 'serviceCategories': [
+#                     'Archiving, backup and disaster recovery', 
+#                     'Compute and application hosting', 
+#                     'Container service', 
+#                     'Content delivery network', 
+#                     'Data warehousing', 
+#                     'Distributed denial of service attack (DDOS) protection', 
+#                     'Firewall', 
+#                     'NoSQL database', 
+#                     'Other database services', 
+#                     'Relational database',
+#                 ], 
+#                 'serviceConstraints': 'None', 
+#                 'serviceDescription': 'Saas', 
+#                 'serviceFeatures': ['one cost delivery', 'money back guarantee'], 
+#                 'serviceName': 'Saas', 
+#                 'standardsCSASTAR': False, 
+#                 'standardsISO28000': False, 
+#                 'standardsISOIEC27001': False, 
+#                 'standardsPCI': False, 
+#                 'status': 'not-submitted', 
+#                 'supplierId': 92277, 
+#                 'supplierName': 'Govdata Ltd', 
+#                 'supportAvailableToThirdParty': True, 
+#                 'supportLevels': 'one off cost', 
+#                 'systemRequirements': ['N/A', 'N/A'], 
+#                 'updatedAt': '2018-05-10T08:22:31.730794Z', 
+#                 'usageNotifications': False, 
+#                 'webChatSupport': 'no', 
+#                 'webInterface': False
+#             }
+#         ]
+#     },
+# ]
+
+# USERS_TEST_CASES = [
+#     {
+#         'users': []
+#     },
+#     {
+#         'users': [
+#             {
+#                 'active': False,
+#                 'emailAddress': 'abc@example.com'
+#             },
+#             {
+#                 'active': True,
+#                 'emailAddress': 'abc@example.com'
+#             },
+#         ]
+#     },
+# ]
+
+MESSAGE_TEST_CASES = (
+    (
+        {
+            'services': []
+        },
+        [
+            {
+                'supplierId': 0,
+                'applicationCompanyDetailsConfirmed': True,
+                'declaration': {'status': 'submitted', 'primaryContactEmail': 'abc@example.com'}
+            },
+            {
+                'supplierId': 1,
+                'applicationCompanyDetailsConfirmed': None,
+                'declaration': {'status': 'submitted', 'primaryContactEmail': 'abc@example.com'}
+            },
+        ],
+        {
+            'users': [
+                {
+                    'active': False,
+                    'emailAddress': 'abc@example.com'
+                },
+                {
+                    'active': True,
+                    'emailAddress': 'abc@example.com'
+                },
+            ]
+        },
+        4,
+        '',
```

Comment:
So to check my understanding, I take it this tuple will eventually have the following scenarios, where emails are sent to each supplier's primary contact, plus the one active user:

- supplier has done 0/3 steps 
- supplier has done 1/3 steps (confirmed company details)
- supplier has done 2/3 steps (confirmed company details and submitted declaration)
- supplier has done 2/3 steps (confirmed company details and completed a service)

And for a supplier who has done 3/3 steps, we will check that 0 emails have been sent?

Reply:
That was a temporary insput to verify that mock/parametrize worked